src/Modules/ContentBlocks/Domain/ContentBlock/Command/CreateContentBlockHandler.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Command;
+
+use ForkCMS\Core\Domain\MessageHandler\CommandHandlerInterface;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlock;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\Block;
+use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\ModuleBlock;
+use ForkCMS\Modules\Internationalisation\Domain\Translation\TranslationKey;
+
+final class CreateContentBlockHandler implements CommandHandlerInterface
+{
+    public function __construct(
+        private readonly BlockRepository $blockRepository,
+        private readonly ContentBlockRepository $contentBlockRepository
+    ) {
+    }
+
+    public function __invoke(CreateContentBlock $createContentBlock)
+    {
+        $createContentBlock->extraId = $this->getNewExtraId($createContentBlock->title);
```

Comment:
the extra id is set here

Reply:
Aha. But how do I get this data from the underlying block (= frontent_block? or page_revision_block?) in the associated wiget?

Reply:
I've added two methods in the controller getSetting and hasSetting that will allow you to get settings from the block